src/App.jsx

```diff
@@ -1,12 +1,89 @@
-import './App.css'
+import React, { useState } from 'react';
+import './App.css';
 
 function App() {
-
+  const [data, setData] = useState([]);
+  const [input, setInput] = useState('');
+  const [edit, setEdit] = useState(null);
+  const handleAddTodo = (e) => {
+    e.preventDefault();
+    if (input !== '') {
+      if (edit !== null) {
+        const id = edit[0].id;
+        setData(data.map((e) => (e.id === id ? { ...e, name: input } : e)));
+        setEdit(null);
+      } else {
+        const newData = [
+          ...data,
+          { name: input, id: Math.floor(Math.random() * 10), check: false },
+        ];
+        setData(newData);
+      }
+      setInput('');
+    }
+  };
+  console.log(data);
+  const handleOnChange = (e) => {
+    setInput(e.target.value);
+  };
+  const handleEdit = (id) => {
+    const newdata = data;
+    const value = newdata.filter((item) => item.id === id);
+    setInput(value[0].name);
+    setEdit(value);
+  };
+  const handleDelete = (id) => {
+    console.log(id);
+    const newdata = [...data];
+    const s =newdata.filter((value)=>value.id!==id)
+    setData(s);
+    setInput('');
+  };
+  const handleCheckboxChange = (id) => {
+    setData(data.map((e)=>e.id===id?{...e,check:!e.check}:e))
+  };
   return (
-    <h1 className="text-3xl font-bold underline text-fuchsia-500	">
-    Hello world!
-  </h1> 
-  )
+    <>
+    <div className='w-screen min-h-screen'>
+      {data.length > 0 && (
```

Comment:
create a separate component for listing todos

Reply:
done